src/domain/image/validation.ts

```diff
@@ -73,10 +73,16 @@ export const addImageSchema = Yup.object().shape(
       [ADD_IMAGE_FIELDS.IMAGE_FILE, ADD_IMAGE_FIELDS.URL],
       validateSelectedImage as any
     ),
-    [ADD_IMAGE_FIELDS.IMAGE_FILE]: Yup.mixed().when(
-      [ADD_IMAGE_FIELDS.SELECTED_IMAGE, ADD_IMAGE_FIELDS.URL],
-      validateFile as any
-    ),
+    [ADD_IMAGE_FIELDS.IMAGE_FILE]: Yup.mixed()
+      .test(
+        'fileNameLength',
+        'image.form.validation.fileNameTooLong',
+        (f) => !f || f.name.length <= 200
+      )
```

Comment:
Not caused by this PR, but after this change the validation of the file will be in quite many different files. 

File type and minimum dimensions are validated at:
https://github.com/City-of-Helsinki/linkedcomponents-ui/blob/e213303a25241e38149a2cb2ec953fb0e4af0734/src/common/components/imageUploader/imageDropzone/ImageDropzone.tsx#L32-L44

And after compressing the file the maximum file size is validated at:
https://github.com/City-of-Helsinki/linkedcomponents-ui/blob/e213303a25241e38149a2cb2ec953fb0e4af0734/src/domain/image/addImageForm/AddImageForm.tsx#L64-L67

Could be nice to have them in one place and also show validation error similar way using toast message or ErrorMessage component